android/src/toga_android/app.py

```diff
@@ -218,14 +218,15 @@ async def intent_result(self, intent):
         :returns: A Dictionary containing "resultCode" (int) and "resultData" (Intent or None)
         :rtype: dict
         """
-        if not intent.resolveActivity(self.native.getPackageManager()):
-            raise RuntimeError("No appropriate Activity found to handle this intent.")
-        self._listener.last_intent_requestcode += 1
-        code = self._listener.last_intent_requestcode
+        try:
+            self._listener.last_intent_requestcode += 1
+            code = self._listener.last_intent_requestcode
 
-        result_future = asyncio.Future()
-        self._listener.running_intents[code] = result_future
+            result_future = asyncio.Future()
+            self._listener.running_intents[code] = result_future
 
-        self.native.startActivityForResult(intent, code)
-        await result_future
-        return result_future.result()
+            self.native.startActivityForResult(intent, code)
+            await result_future
+            return result_future.result()
+        except AttributeError:
```

Comment:
Have you tested if this AttributeError is raised when there is no appropriate Activity? If it does, then that's fine. Otherwise, you could use `except Exception`

Reply:
AttributeError is what is raised when no valid activity is given. Tried it on a couple of valid and all worked and then threw random stuff at it and it always errored out with AttributeError

Reply:
On Android, BeeWare has now switched from Rubicon to Chaquopy, which has full support for Java exceptions. So if this is working, it may only be by accident.

It's not obvious to me where an AttributeError would be coming from. Can you give some examples of the valid and invalid intents you used to test this, and the AttributeErrors you received?

Reply:
For valid intents I used the intent given in the linked issue as well as ACTION_GET_CONTENT for sure and then a few other random intents. For invalid ones I put in spelling errors as well as random strings of characters. 

The attribute error only returns if it can't find the given intent to it would probably be good to use the proper error to catch it Incase this is just luck that it is working. I can look at getting the error catch changed soon